### DISCUSSION TITLE
There is anyway to plug some APM service on Directus?

### DISCUSSION THREAD

**Discussion by gabrsar** [directus/directus#12626] [Archive]
I wish I could see some APM metrics from my directus instance. In my case I use NewRelic and the only way I got to make it start was by injecting the `require("newrelic")` it into `/directus/node_modules/directus/dist/server.js`. :eyes: 

This makes it work really so so, as it is not able to split transactions by URLs and so on. There is any one with some experience doing it or any other APM tool? Or a correct way of doing it?

Thank you.

**Comment by rijkvanzanten** [comment]
You could use one of the `init` hooks to register global middleware that fires on every request, or potentially use the `response` action hook to log something on every request 🙂

**Comment by gabrsar** [comment]
Just posting here to help any other soul.

A better way that I found to plug Elastic APM or NewRelic into Directus was to add it on the FIRST line of `/directus/node_modules/directus/dist/server.js` instead of at the end (as NewRelic doc tells me to do). This was a gamechanger for NewRelic mapping and so on, everything started to work as expected. 

If there is any way to add a customizable call before directus even starts to make things like this without all the mess I had to do, would be really nice.

**Reply by kwarpechowski** [reply]
@gabrsar I had the same issue.
Extend server.js is not possible. But, I resigned from start directus via npx. I created custom file, who include new relic and run directus via code.

```
require('dotenv').config();
require('newrelic');
const { startServer } = require('directus/server');
startServer();
```


in package.json

```
  "scripts": {
    "start": "node scripts/start.js"
  },
```

**Reply by nobleach** [reply]
I'm trying to figure out how to do this after Directus made the change to the new `@directus` namespace. The exports seem quite a bit different.
